packages/xo-web/src/xo-app/logs/index.js

```diff
@@ -119,31 +121,40 @@ const UNHEALTHY_VDI_CHAIN_ERROR = 'unhealthy VDI chain'
 const UNHEALTHY_VDI_CHAIN_LINK =
   'https://xen-orchestra.com/docs/backup_troubleshooting.html#vdi-chain-protection'
 
+const filterOptionRenderer = ({ label, value }) =>
+  _(label, message => <option value={value}>{message}</option>)
+
 class Log extends BaseComponent {
   state = {
-    filter: 'all',
+    filter: DEFAULT_CALL_FILTER,
   }
 
-  render () {
-    const { props, state } = this
-    const predicate = PREDICATES[state.filter]
+  _getFilteredCalls = () =>
+    filter(this.props.log.calls, PREDICATES[this.state.filter.value])
+
+  _filterValueRenderer = ({ label, value }) =>
+    _(label, message => (
+      <option value={value}>
+        {message} ({this._getFilteredCalls().length})
```

Comment:
This will trigger a filter on each render. Use a selector.

Reply:
no need here, static data

Reply:
Right, but if you create a selector, better use it everywhere :wink: